src/Persistence.php

```diff
@@ -429,14 +429,15 @@ public function _serializeLoadField(Field $f, $value)
         case 'serialize':
             return unserialize($value);
         case 'json':
-            switch ($f->type) {
-            case 'array':
-                return json_decode($value, true);
-            case 'object':
-                return json_decode($value, false);
+            if (!defined('JSON_THROW_ON_ERROR')) {
+                define('JSON_THROW_ON_ERROR', 0);
+            }
+            $res = json_decode($value, $f->type == 'array', 512, JSON_THROW_ON_ERROR);
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                throw new \Exception(json_last_error_msg());
```

Comment:
You should better throw \atk4\data\Exception here and also add json_last_error() in there.
```
throw new Exception([
    'There was error while decoding JSON
    'field' => $f->name,
    'code' => json_last_error(),
    'error' => json_last_error_msg(),
]);
```

Reply:
i feel that Json validation is not related to data exception,which is why i used \Exception.

Reply:
well.... ok.